lit-mar/lambda_function.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import print_function
+
+import urllib
+import sys
+import os
+import json
+import subprocess
+
+class Context(object):
+    def __init__(self, logger, metrics_writter, request, response, system_info):
+        self.logger = logger
+        self.metrics_writter = metrics_writter
+        self.request = request
+        self.response = response
+        self.system_info = system_info
+
+
+class BaseResponse(object):
+    """BaseResponse if the object providing standard response attributes and operations"""
+    def __init__(self, response_body=""):
+        """Constructor, construct a response
+
+        Parameters
+        ----------
+        response_body: Optionally give the response a non-empty body
+        """
+        self.headers = []
+        self.status_code = 200
+        self.body = response_body
+
+    def get_dict(self):
+        """Get the response in dict form"""
+        return {"headers":dict(self.headers), "body":self.body, "statusCode":self.status_code}
+
+    def add_response_property(self, key, value):
+        """Add a pair of property to response
+
+        Parameters
+        ----------
+        key: string
+            Key of the property
+        value: string
+            Value of the property
+        """
+        self.headers.append((key, value))
+
+    def set_response_status(self, code, reason_string=None):
```

Comment:
please remove the explicit setter methods. Use constructor instead to set these values. Or use Python's `@property` - See: http://powerfulpython.com/blog/python-properties-refactoring/

Reply:
This is to meet MAR spec. See here https://corp.quip-amazon.com/6PrOAOQTBBHE

Reply:
see comment above